src/par_iter/mod.rs

```diff
@@ -20,44 +20,49 @@ use self::enumerate::Enumerate;
 use self::filter::Filter;
 use self::filter_map::FilterMap;
 use self::flat_map::FlatMap;
-use self::from_par_iter::FromParallelIterator;
 use self::map::{Map, MapFn, MapCloned, MapInspect};
 use self::reduce::{reduce, ReduceOp, SumOp, ProductOp, ReduceWithIdentityOp, SUM, PRODUCT};
 use self::skip::Skip;
 use self::take::Take;
-use self::internal::*;
 use self::weight::Weight;
 use self::zip::ZipIter;
 
+pub use self::from_par_iter::FromParallelIterator;
 pub use self::string::ParallelString;
 
-pub mod find;
-pub mod find_first_last;
-pub mod chain;
-pub mod collect;
-pub mod enumerate;
-pub mod filter;
```

Comment:
While I definitely think we should make these modules private, I think we also need to continue to publicly export types like `par_iter::filter::Filter` -- that is, the return types of iterator adapters, much like the corresponding types in libstd are public. This perhaps amounts to making things like `use self::filter::Filter` above into `pub use`.